spec/components/concern/has_custom_fields_spec.rb

```diff
@@ -31,6 +31,33 @@ class CustomFieldsTestItemCustomField < ActiveRecord::Base
       Object.send(:remove_const, :CustomFieldsTestItemCustomField)
     end
 
+    it "allows preloading of custom fields" do
+      test_item = CustomFieldsTestItem.new
+      CustomFieldsTestItem.preload_custom_fields([test_item], ["test_field"])
+      expect(test_item.preloaded_custom_fields).to eq({ "test_field" => nil })
+    end
+
+    it "errors if a custom field is not preloaded" do
+      test_item = CustomFieldsTestItem.new
+      CustomFieldsTestItem.preload_custom_fields([test_item], ["test_field"])
+      expect { test_item.custom_fields["other_field"] }.to raise_error(HasCustomFields::NotPreloadedError)
+    end
+
+    it "resets the preloaded_custom_fields if preload_custom_fields is called twice" do
+      test_item = CustomFieldsTestItem.new
+      CustomFieldsTestItem.preload_custom_fields([test_item], ["test_field"])
+      CustomFieldsTestItem.preload_custom_fields([test_item], ["other_field"])
+      expect(test_item.preloaded_custom_fields).to eq({ "other_field" => nil })
+    end
+
+    it "resets preloaded proxy for custom fields if preload_custom_fields is called twice" do
```

Comment:
preloaded proxy is an internal implementation detail of the module itself which I do not think should be exposed even as a description in the test. From the assertion, we're testing the `#custom_fields` instance method so I think the description should reflect that.

Reply:
Ok fixing